src/management-system-v2/lib/data/legacy/iam/users.ts

```diff
@@ -78,7 +78,9 @@ export function deleteuser(userId: string) {
   if (!user) throw new Error("User doesn't exist");
 
   for (const environmentId of Object.keys(environmentsMetaObject)) {
-    if (environmentsMetaObject[environmentId].ownerId === userId) deleteEnvironment(environmentId);
+    const environment = environmentsMetaObject[environmentId];
+    if ((!environment.organization || environment.active) && environment.ownerId === userId)
+      deleteEnvironment(environmentId);
```

Comment:
What is this supposed to check? It seems like currently an organization that is active would be true?

Reply:
`(!environment.organization || environment.active) `  selects the personal environments, and all active organization environments, in theory a user's ID will only be on an environment (ownerId) if it is active, but this is just a way to keep typescript happy (Inactive environments don't have an ownerId), maybe a better way would've been to check if the environment has a owenrId, but this logic will be replaced soon either way.